### PR TITLE
#162948895 Test Driven Development

### PR DESCRIPTION
**Pull request by tesh254**
#### What does this PR do?

This pr contains  a couple of tests, it approaches the TDD way

#### Description of the task to be accomplished?

* Create a couple of test for authentication

#### How can this be manually tested?

* Clone this repo
* CD into the project directory
* Install dependencies `pip install -r requirements.txt`
* Run this command to run tests `python3 manage.py test`

#### Background context?

* Use Python3 version `3.7 or above 3.x`
* Use Django version `2.x`

#### Related Pivotal Tracker stories?

* [#162948895](https://www.pivotaltracker.com/story/show/162948895)

### CODE REVIEW

**Review by jamesbeamie — APPROVED** [review]

**Review by Ruiru11 — APPROVED** [review]

**Review by danuluma — APPROVED** [review]

**Comment by boltonotieno** [comment]
The command for running tests is not correct. It should be `python3 manage.py test`

**Comment by tesh254** [comment]
Fixed

File: authors/apps/authentication/tests/test_registration.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        }
+
+    def test_user_registration(self):
+        """Test the user registration"""
+        response = self.client.post(
+            self.register_url,
+            self.valid_user_credentials,
+            format="json"
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
```
Comment (frankip) [inline authors/apps/authentication/tests/test_registration.py:51]: @tesh254 you need to also assert for the status messages

**Review by frankip — CHANGES_REQUESTED** [review]

**Review by frankip — APPROVED** [review]

**Review by frankip — APPROVED** [review]
LGTM

**Pull request merged**